### PR TITLE
Bump up golang to 1.12.12

### PR DESCRIPTION
**Pull request by reasonerjt**
Bump up the golang for compiling the binaries to 1.12.12

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: make/photon/registry/Dockerfile.binary [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.11
+FROM golang:1.12.12
```
Comment (wy65701436) [inline make/photon/registry/Dockerfile.binary]: The version of golang is just referring to the upstream, do we really need to upgrade it?

Same as Notary and Clair, please double confirm.

File: tests/travis/api_common_install.sh [hunk 2]
```diff
@@ -24,5 +24,5 @@ sudo curl -o /home/travis/gopath/src/github.com/goharbor/harbor/tests/apitests/p
 sudo apt-get update && sudo apt-get install -y --no-install-recommends python-dev openjdk-7-jdk libssl-dev && sudo apt-get autoremove -y && sudo rm -rf /var/lib/apt/lists/*
 sudo wget https://bootstrap.pypa.io/get-pip.py && sudo python ./get-pip.py && sudo pip install --ignore-installed urllib3 chardet requests && sudo pip install robotframework==3.0.4 robotframework-httplibrary requests dbbot robotframework-pabot --upgrade
 sudo make swagger_client
-sudo make install GOBUILDIMAGE=golang:1.12.5 COMPILETAG=compile_golangimage CLARITYIMAGE=goharbor/harbor-clarity-ui-builder:1.6.0 NOTARYFLAG=true CLAIRFLAG=true CHARTFLAG=true
+sudo make install GOBUILDIMAGE=golang:1.12.122 COMPILETAG=compile_golangimage CLARITYIMAGE=goharbor/harbor-clarity-ui-builder:1.6.0 NOTARYFLAG=true CLAIRFLAG=true CHARTFLAG=true
```
Comment (wy65701436) [inline tests/travis/api_common_install.sh]: 1.12.122? typo?

File: make/photon/registry/Dockerfile.binary (hunk 1 above)
Comment (reasonerjt) [reply make/photon/registry/Dockerfile.binary]: I think yes, we need to patch the artifacts we build using golang.  Any disagreement?

File: tests/travis/api_common_install.sh (hunk 2 above)
Comment (reasonerjt) [reply tests/travis/api_common_install.sh]: it is... Thanks

**Pull request merged**